agent/bench-scripts/pbench-linpack

```diff
@@ -34,101 +24,167 @@ threads=`cat /proc/cpuinfo | grep processor | wc -l`
+	printf -- "\t       --samples=<int>      number of samples to use per test iteration (default is ${def_nr_samples})\n"
+	printf -- "\t       --threads=int[,int]  number of threads to use (default is # local CPUs)\n"
 	printf -- "\t       --tool-group=str\n"
 	printf -- "\t       --sysinfo=str,       str= comma separated values of sysinfo to be collected\n"
 	printf -- "\t                                available: $(pbench-display-sysinfo-options)\n"
 }
 
 # Process options and arguments
-opts=$(getopt -q -o C:h --longoptions "config:,threads:,tool-group:,sysinfo:,help" -n "getopt.sh" -- "$@");
-if [ $? -ne 0 ]; then
-	printf -- "${script_name} $*\n"
+opts=$(getopt -q -o C:h --longoptions "config:,samples:,threads:,tool-group:,sysinfo:,help" -n "getopt.sh" -- "${@}")
+if [[ ${?} -ne 0 ]]; then
+	printf -- "${script_name} ${*}\n"
 	printf -- "\n"
 	printf -- "\tunrecognized option specified\n\n"
 	usage
 	exit 1
 fi
-eval set -- "$opts";
+eval set -- "${opts}"
 while true; do
-	case "$1" in
-		-c|--config)
-		shift;
-		if [ -n "$1" ]; then
-			config="$1"
+	arg=${1}
+	shift
+	case "${arg}" in
+	-c|--config)
+		if [[ -n "${1}" ]]; then
+			config="${1}"
 			shift;
 		fi
 		;;
-		--threads)
-		shift;
-		if [ -n "$1" ]; then
-			threads="$1"
+	--samples)
+		if [[ -n "${1}" ]]; then
+			nr_samples="${1}"
 			shift;
 		fi
 		;;
-		--tool-group)
-		shift;
-		if [ -n "$1" ]; then
-			tool_group="$1"
+	--threads)
+		if [[ -n "${1}" ]]; then
+			threads="${1}"
 			shift;
 		fi
 		;;
-		--sysinfo)
-		shift;
-		if [ -n "$1" ]; then
-			sysinfo="$1"
+	--tool-group)
+		if [[ -n "${1}" ]]; then
+			tool_group="${1}"
+			shift;
+		fi
+		;;
+	--sysinfo)
+		if [[ -n "${1}" ]]; then
+			sysinfo="${1}"
 			shift;
 		fi
 		;;
-		-h|--help)
+	-h|--help)
 		usage
 		exit 0
 		;;
-		--)
+	--)
 		shift;
 		break;
 		;;
-		*)
-		echo "what happened? [$1]"
-		exit 0
-		break;
+	*)
+		echo "what happened? [${1}]" >&2
+		exit 1
 		;;
 	esac
 done
-verify_common_bench_script_options $tool_group $sysinfo
+verify_common_bench_script_options ${tool_group} ${sysinfo}
+
+ver="$(pbench-config version ${benchmark})"
+if [[ -z "${ver}" ]]; then
+    error_log "${script_name}: package version is missing in config file"
+    exit 1
+fi
+if [[ -z "${linpack_dir}" ]]; then
+    linpack_dir="/usr/local/${script_name}-${ver}/benchmarks/linpack"
+    linpack_dir_kind="default"
+else
+    linpack_dir_kind="provided"
+fi
+if [[ ! -d "${linpack_dir}" ]]; then
+    error_log "${script_name}: the ${linpack_dir_kind} linpack directory, ${linpack_dir}, does not exist"
+    exit 1
+fi
+linpack_cmd="${linpack_dir}/xlinpack_xeon64"
+if [[ ! -x "${linpack_cmd}" ]]; then
+    error_log "${script_name}: the expected linpack command, ${linpack_cmd}, does not exist"
+    exit 1
+fi
 
-## Ensure the right version of the benchmark is installed
-check_install_rpm "${benchmark_rpm}" "${ver}"
+function store_and_run {
+	local workdir=${1}
+	local cmd=${2}
+	local filename=${3}
+	(
+		cd "${1}" &&
```

Comment:
Shouldn't that be
```
		cd "${workdir}" &&
```
🙃

Reply:
Fixed in https://github.com/ldoktor/pbench/pull/5.